projects/angular/src/data/tree-view/tree-node.ts

```diff
@@ -188,6 +199,20 @@ export class ClrTreeNode<T> implements OnInit, OnDestroy {
     this.subscriptions.push(
       this._model.loading$.pipe(debounceTime(0)).subscribe(isLoading => (this.isModelLoading = isLoading))
     );
+
+    this.subscriptions.push(
+      this.typeAheadKeyEvent.pipe(debounceTime(TREE_TYPE_AHEAD_TIMEOUT)).subscribe((bufferedKeys: string) => {
```

Comment:
Wonder if we could skip this subscription if we're not using the type ahead functionality.
On a tree with 10,000 nodes, this will be 10,000 subscriptions less.